extensions/panache/panache-rest-hibernate-orm/deployment/src/main/java/io/quarkus/panache/rest/hibernate/orm/deployment/EntityDataAccessImplementor.java

```diff
@@ -30,6 +31,16 @@ public ResultHandle listAll(BytecodeCreator creator) {
         return creator.invokeStaticMethod(ofMethod(entityClassName, "listAll", List.class));
     }
 
+    @Override
+    public ResultHandle list(BytecodeCreator creator, ResultHandle limit) {
```

Comment:
Maybe a comment on how this is implemented. The `-1` is kind of confusing

Reply:
Sure I'll add it.

Basically it's the terminology difference. Limit assumes how many elements do you expect. But Panache `range` method takes ranges starting with 0. So if you want one element, you need ranges from 0 to 0.

Reply:
:+1: 

That should go in the comments :)